projects/components/src/datagrid/datagrid.component.html

```diff
@@ -1,4 +1,60 @@
+<ng-template #dropdown>
+    <clr-dropdown class="btn-group-overflow open" *ngIf="buttonConfig.contextualButtons.buttons.length !== 0">
+        <button class="btn dropdown-toggle dropdown-small" clrDropdownTrigger>
+            <clr-icon shape="ellipsis-horizontal"></clr-icon>
+        </button>
+        <clr-dropdown-menu class="dropdown-menu" *clrIfOpen>
+            <ng-container *ngFor="let button of buttonConfig.contextualButtons.buttons">
+                <button
+                    class="btn"
+                    [ngClass]="button.class"
+                    [class.disabled]="!button.shouldDisplay(getDatagridSelection())"
+                    (click)="button.handler(getDatagridSelection())"
+                >
+                    {{ button.label }}
+                </button>
+            </ng-container>
+        </clr-dropdown-menu>
+    </clr-dropdown>
+</ng-template>
+
 <clr-datagrid [clrDgLoading]="isLoading" [ngClass]="this.clrDatagridCssClass" (clrDgRefresh)="gridStateChanged($event)">
+    <clr-dg-action-bar *ngIf="buttonConfig">
+        <div class="btn-group" *ngFor="let button of buttonConfig.globalButtons">
+            <button class="btn" [ngClass]="button.class" *ngIf="button.shouldDisplay()" (click)="button.handler()">
+                {{ button.label }}
+            </button>
+        </div>
+
+        <ng-container
+            *ngIf="
+                buttonConfig.contextualButtons.position === ContextualButtonPosition.TOP &&
+                getDatagridSelection().length !== 0
+            "
+        >
+            <div class="btn-group" *ngIf="this.getFeaturedButtons(this.getDatagridSelection()).length !== 0">
+                <ng-container *ngFor="let button of this.getFeaturedButtons(this.getDatagridSelection())">
+                    <button type="button" class="btn btn-icon" (click)="button.handler(getDatagridSelection())">
```

Comment:
The signature of `getDatagridSelection` method in the TS below can be changed to `get datagridSelection`. This way calls to the method can be a little less verbose as `button.handler(datagridSelection)`